src/main/java/mods/railcraft/common/plugins/forge/WorldPlugin.java

```diff
@@ -140,12 +140,11 @@ public static boolean playerRemoveBlock(World world, BlockPos pos, @Nullable Ent
         if (MinecraftForge.EVENT_BUS.post(new BlockEvent.BreakEvent(world, pos, state, player)))
             return false;
 
-        if (state.getBlock().removedByPlayer(state, world, pos, player, dropBlock))
+        if (!state.getBlock().removedByPlayer(state, world, pos, player, dropBlock))
             return false;
 
         if (dropBlock) {
-            List<ItemStack> drops = state.getBlock().getDrops(world, pos, state, 0);
-            InvTools.dropItems(drops, world, pos);
+            state.getBlock().harvestBlock(world, player, pos, state, getBlockTile(world, pos), player.getHeldItem(player.getActiveHand()));
```

Comment:
I don't know what you are trying to do here, but the tile no longer exists at this point, so calling getBlockTile(world, pos) will return null always.